src/systems/a320_systems/src/hydraulic.rs

```diff
@@ -1,16 +1,879 @@
+
+            //UPDATING ACTUATOR PHYSICS AT "FIXED STEP / ACTUATORS_SIM_TIME_STEP_MULT"
+            //Here put everything that needs higher simulation rates
+            let num_of_actuators_update_loops =
+                num_of_update_loops * A320Hydraulic::ACTUATORS_SIM_TIME_STEP_MULT;
+            let delta_time_physics =
+                min_hyd_loop_timestep / A320Hydraulic::ACTUATORS_SIM_TIME_STEP_MULT; //If X times faster we divide step by X
+            for _cur_loop in 0..num_of_actuators_update_loops {
+                self.rat
+                    .update_physics(&delta_time_physics, &ct.indicated_airspeed);
+            }
+        }
+    }
+
+    pub fn update_hyd_logic_inputs(
+        &mut self,
+        delta_time_update: &Duration,
+        overhead_panel: &A320HydraulicOverheadPanel,
+        ct: &UpdateContext,
+    ) {
+        let mut cargo_operated_ptu = false;
+        let mut cargo_operated_ypump = false;
+        let mut nsw_pin_inserted = false;
+
+        //Only evaluate ground conditions if on ground, if superman needs to operate cargo door in flight feel free to update
+        if self.hyd_logic_inputs.weight_on_wheels {
+            cargo_operated_ptu = self
+                .hyd_logic_inputs
+                .is_cargo_operation_ptu_flag(&delta_time_update);
+            cargo_operated_ypump = self
+                .hyd_logic_inputs
+                .is_cargo_operation_flag(&delta_time_update);
+            nsw_pin_inserted = self
+                .hyd_logic_inputs
+                .is_nsw_pin_inserted_flag(&delta_time_update);
+        }
+
+        //Basic faults of pumps //TODO wrong logic: can fake it using pump flow == 0 until we implement check valve sections in each hyd loop
+        //At current state, PTU activation is able to clear a pump fault by rising pressure, which is wrong
+        if self.yellow_electric_pump.is_active() && !self.is_yellow_pressurised() {
+            self.hyd_logic_inputs.yellow_epump_has_fault = true;
+        } else {
+            self.hyd_logic_inputs.yellow_epump_has_fault = false;
+        }
+        if self.engine_driven_pump_1.is_active() && !self.is_green_pressurised() {
+            self.hyd_logic_inputs.green_edp_has_fault = true;
+        } else {
+            self.hyd_logic_inputs.green_edp_has_fault = false;
+        }
+        if self.engine_driven_pump_2.is_active() && !self.is_yellow_pressurised() {
+            self.hyd_logic_inputs.yellow_edp_has_fault = true;
+        } else {
+            self.hyd_logic_inputs.yellow_edp_has_fault = false;
+        }
+        if self.blue_electric_pump.is_active() && !self.is_blue_pressurised() {
+            self.hyd_logic_inputs.blue_epump_has_fault = true;
+        } else {
+            self.hyd_logic_inputs.blue_epump_has_fault = false;
+        }
+
+        //RAT Deployment //Todo check all other needed conditions
+        if !self.hyd_logic_inputs.eng_1_master_on
+            && !self.hyd_logic_inputs.eng_2_master_on
```

Comment:
More encapsulation examples:

```suggestion
        if self.hyd_logic_inputs.neither_engine_master_is_on()
```